inf_post/include/balanced.h

```diff
@@ -0,0 +1,4 @@
+#include <stack>
```

Comment:
Why have you written down each function in a separate file????? They all are checks. combine them

Reply:
+